src/pages/events.tsx

```diff
@@ -6,8 +6,16 @@ import EventsList from '../components/EventsList';
 
 const EventsPage: React.FC<PageProps> = () => {
   const currDate = new Date().setHours(0, 0, 0, 0);
-  const upcomingEvents = EVENTS.filter(({ date }: Event) => date.getTime() >= currDate);
-  const pastEvents = EVENTS.filter(({ date }: Event) => date.getTime() < currDate);
+  const { upcomingEvents, pastEvents } = React.useMemo(() => {
```

Comment:
seems like unnecessary optimization to use `useMemo` here. the events data is static which guarantees no re-renders; `currDate` is setup only once on page load. nothing warrants to use `useMemo` in this case

Reply:
Oh yeah I thought that maybe in the future we would dynamically add events based off of like scraping off NYU Engage but also I don't really know how React does this, let me change that, but what do you think about the sorting here?

Reply:
yeah, we should sort it here just like you did it. I think that is a really good addition.

regarding the memoization, your intent makes sense but I would say that it's not worth optimizing unless you see poor performance. the `useMemo` hook basically prevents the wrapped value from recomputing between renders if the dependencies don't change. renders are caused by any state changes, and there is no state in the events page component. also, the sorting and filtering computation is relatively cheap -- there are only like 15 events or so. let me know if it doesn't make sense

Reply:
Yeah that makes sense. Does `useMemo` use more memory to do that? Like I now can understand why `useMemo` doesn't do anything here, but are there any performance reasons for not having it, or is it mostly for readability?

Reply:
nvm just gonna google it... ty for the review!

Reply:
honestly, in this case the performance differences will be negligible. it wouldn't make a difference even if there was `useMemo`... but it makes the code look cluttered without a justifiable reason.

here's a good article about it's performance footprint https://filipkastovsky.github.io/blog/usememo-considered-harmful/